src/Calculator.cpp

```diff
@@ -188,16 +195,45 @@ Result<std::stack<std::string>> Calculator::infixToPostfix(std::string infixStri
                 break;
 
             case '+' :
-                //Fallthrough
-            // FIXME allow '-' to be used to denote negative numbers
-            case '-' :
                 while (!operatorStack.empty() && operatorStack.top() != '(')
                 {
                     outputStack.push(std::string(1, operatorStack.top()));
                     operatorStack.pop();
                 }
                 operatorStack.push(infixString[i]);
                 break;
+            case '-' :
+                // If the previous symbol isn't part of a number or a right parenthesis
+                if (i == 0 || !(isdigit(infixString[i - 1]) || infixString[i - 1] == '.' || infixString[i - 1] == ')'))
```

Comment:
This case statement is becoming pretty long. Maybe it should be pulled out into a separate helper method.